harvester/create.go

```diff
@@ -46,6 +48,31 @@ func (d *Driver) Create() error {
 		if err = json.Unmarshal([]byte(d.VMAffinity), &affinity); err != nil {
 			return err
 		}
+		//VM naming convention is of form: clusterName-poolName-generatedString
+		//we can reverse split this to identify unique machinesets name, to label nodes
+		//with this unique machine set. This can then be used for populating affinity rules
+		machineSetSplit := strings.Split(d.MachineName, "-")
+		machineSetName := strings.Join(machineSetSplit[:len(machineSetSplit)-1], "-")
```

Comment:
It looks like the machine name is `clusterName-poolName-generatedString-randomString`. In my cluster, the machine is `rke2-3-pool1-e0b8749d-g9jln`, `rke2-3-pool1-e0b8749d-r96bc`, and `rke2-3-pool1-e0b8749d-wrbf8`. Do we want to use `strings.Join(machineSetSplit[:len(machineSetSplit)-2], "-")`?

Reply:
we could use either one, but i need to check if it is feasible to have same pool names

Reply:
Yeah, there may have conflicts if users create clusters with the same name in different namespaces. Probably, we can use `<cluster-namespace>-<cluster-name>-<pool-name>`.

Reply:
yeah that might work.. I will make the change.

Reply:
PR now includes additional logic to drop generatedID from pool name and includes namespace.